instrumentation/aws-sdk/aws-sdk-2.2/javaagent/src/test/java/io/lumigo/javaagent/instrumentation/awssdk/v2_2/Aws2SqsTracingTest.java

```diff
@@ -0,0 +1,311 @@
+    ReceiveMessageResponse response =
+        client.receiveMessage(builder -> builder.queueUrl(queueUrl).build());
+    assertThat(response.messages()).hasSize(1);
+    Condition<Message> messageCondition =
+        new Condition<>(
+            message -> message.body().equals(messageBody), "message body equals " + messageBody);
+    assertThat(response.messages().get(0)).has(messageCondition);
+
+    instrumentation.waitAndAssertTracesWithoutScopeVersionVerification(
+        trace ->
+            trace
+                .hasSize(1)
+                .hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("Sqs.CreateQueue")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasNoParent()
+                            .hasAttributesSatisfying(
+                                attrs -> {
+                                  assertThat(attrs)
+                                      .containsEntry(
+                                          AttributeKey.stringKey("aws.agent"), "java-aws-sdk");
+                                  assertThat(attrs)
+                                      .containsEntry(
+                                          AttributeKey.stringKey("aws.queue.name"), queueName);
+                                  assertThat(attrs)
+                                      .containsEntry(
+                                          AttributeKey.stringKey("aws.requestId"),
+                                          "00000000-0000-0000-0000-000000000000");
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.RPC_SYSTEM, "aws-api");
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.RPC_SERVICE, "Sqs");
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.RPC_METHOD, "CreateQueue");
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.HTTP_METHOD, "POST");
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.HTTP_STATUS_CODE, 200L);
+                                  assertThat(attrs)
+                                      .containsEntry(SemanticAttributes.NET_PEER_NAME, "localhost");
+                                  assertThat(attrs)
+                                      .containsEntry(
+                                          SemanticAttributes.NET_PEER_PORT, sqsServerPort);
+                                  assertThat(attrs)
+                                      .containsEntry(
+                                          TracingExecutionInterceptor.HTTP_RESPONSE_BODY_KEY,
+                                          createQueueResponse);
```

Comment:
Queue producers/ consumers add messaging payload?
but the other requests (create queue/ delete queue/ etc..) adds http_response_body?

did I get it right?

Reply:
That's correct.

As the SDK communicates with an HTTP client, those calls add an HTTP Request/Response body.

For SQS (messages), we add a message payload. Right now there isn't specific handling in the OTeL instrumentation for EventBridge or SNS, so we may want to add that too